src/pages/projectSettings/tabs/NotificationsTab/NotificationsRow.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import styled from "@emotion/styled";
+import { NotificationMethod } from "components/Notifications/NotificationMethod";
+import { NotificationSelect } from "components/Notifications/NotificationSelect";
+import { triggers } from "constants/triggers";
+import {
+  SubscriptionMethods,
+  useNotificationModal,
+} from "hooks/useNotificationModal";
+
+import {
+  SUBSCRIPTION_EMAIL,
+  SUBSCRIPTION_JIRA_COMMENT,
+  SUBSCRIPTION_SLACK,
+} from "types/subscription";
+import { Trigger } from "types/triggers";
+import { validateEmail, validateJira, validateSlack } from "utils/validators";
+
+export const subscriptionMethodControls: SubscriptionMethods = {
+  "jira-comment": {
+    label: "JIRA Issue",
+    placeholder: "ABC-123",
+    targetPath: "jira-comment",
+    validator: validateJira,
+  },
+  email: {
+    label: "Email Address",
+    placeholder: "someone@example.com",
+    targetPath: "email",
+    validator: validateEmail,
+  },
+  slack: {
+    label: "Slack Username or Channel",
+    placeholder: "@user",
+    targetPath: "slack",
+    validator: validateSlack,
+  },
+};
+
+export const subscriptionMethodDropdownOptions = [
+  SUBSCRIPTION_JIRA_COMMENT,
+  SUBSCRIPTION_SLACK,
+  SUBSCRIPTION_EMAIL,
+];
+
+interface SubscriptionSelectProps {
+  resourceType;
+  trigger: Trigger;
+  // todo: use in future work
+  selectors;
+  regexSelectors;
+}
+
+export const SubscriptionSelect: React.VFC<SubscriptionSelectProps> = ({
+  trigger,
+}) => {
+  const type = "version";
+  const resourceId = "project-id";
+
+  const {
+    disableAddCriteria,
+    extraFieldInputVals,
+    extraFields,
+    onClickAddRegexSelector,
+    regexSelectorProps,
+    setExtraFieldInputVals,
+    setSelectedTriggerIndex,
+    showAddCriteria,
+    buildInitiatorSelected,
+    setBuildInitiatorSelected,
+  } = useNotificationModal({
+    subscriptionMethodControls,
+    triggers,
+    resourceId,
+    type,
+  });
+  const currentIndex = trigger.index || 0;
+
+  return (
+    <RowContainer>
+      <NotificationSelect
+        triggers={triggers}
+        extraFields={extraFields}
+        selectedTriggerIndex={currentIndex}
+        showAddCriteria={showAddCriteria}
+        setSelectedTriggerIndex={setSelectedTriggerIndex}
+        extraFieldInputVals={extraFieldInputVals}
+        setExtraFieldInputVals={setExtraFieldInputVals}
+        regexSelectorProps={regexSelectorProps}
+        disableAddCriteria={disableAddCriteria}
+        onClickAddRegexSelector={onClickAddRegexSelector}
+        onChangeSelectedOption={setBuildInitiatorSelected}
+        selectedOption={buildInitiatorSelected}
+      />
+    </RowContainer>
+  );
+};
+
+export const SubscriptionMethod = () => {
```

Comment:
Possibly intentional since this is the first part of the ticket, but just wanted to point out that this isn't being used anywhere

If it is going to be used in `SubscriptionField.tsx`, then I'm not sure how I feel about these "wrapper" components. We could alternatively call the hook `useNotificationModel` once in `SubscriptionField.tsx` and get all the information we need in one go, rather than twice across two components

Reply:
good point. I leave this comment open and keep it in mind for the next PR.